tests/mma/mmatest.f90

```diff
@@ -0,0 +1,205 @@
+!        TEST_CASE_2:                                                       !
+!    Problem 1 in "https://people.kth.se/~krille/originalmma.pdf"           !
+!                                                                           !
+!                                                                           !
+! ------------------------------------------------------------------------- !
+!                                                                           !
+! To switch between the above two test cases:                               !
+! uncomment/comment the lines 36-37 and 40-41.                              !
+! uncomment/comment the lines 59-66 and 76-83.                              !
+! uncomment/comment the lines 106-111 and 115-120.                          !
+! Also, in file mma.f90 in "subroutine mma_init(this, x, n, m)",            ! 
+! uncomment/comment lines 166-172 and 175-181.                              !
+!===========================================================================!
+
+
+    use num_types
+    use mma
+    implicit none
+
+
+
+    !!!!!!! TEST_CASE_1
+    integer, parameter :: n =5
+    integer, parameter :: m =1 
+
+    !!!!!!! TEST_CASE_2
+    ! integer, parameter :: n =3
+    ! integer, parameter :: m =2
+
+    integer :: iter, i, j
+    real(kind=rp), dimension(n) :: x
+    real(kind=rp), dimension(m) :: fval
+    real(kind=rp), dimension(m,n) :: dfdx
+    real(kind=rp) :: f0val
+    real(kind=rp), dimension(n) :: df0dx
+    integer :: maxiter
+    logical :: mmadone
+    type(mma_t) :: optprob
+
+    !!!!!!! TEST_CASE_1
+    !!!!!!! for test problem 1 in "https://people.kth.se/~krille/originalmma.pdf" 
+    !!!!!!! change the values of n and m at the begining to 5 and 1, respectively.
+    !!!!!!! integer, parameter :: n =5
+    !!!!!!! integer, parameter :: m =1   
+    ! ! x = [6.0, 5.0, 4.0, 3.0, 2.0]
+    x = [5.0, 5.0, 5.0, 5.0, 5.0]
+    !x =[6.0093,    5.3217 ,   4.5247  ,  3.4820  ,  2.1137]
+    !!!!!! opt value
+    !!!!!! x = (/6.016, 5.309, 4.494, 3.502, 2.153/)
+    call func1 (n, m, x, f0val , df0dx, fval , dfdx)
+    print *, "f0val=", f0val, "fval=", fval
+    print '(A,I3,A,5F10.4,A,F10.7,A,F10.7)', 'iter=', 0, ' , x= ', x, &
+        '-------,f0val= ', f0val, ',   fval= ', fval
+
+
+
+    !!!!!!! TEST_CASE_2
+    !!!!!!! For test problem 1 in:
+    !!!!!!! "https://comsolyar.com/wp-content/uploads/2020/03/gcmma.pdf"
+    !!!!!!! change the values of n and m at the begining to 3 and 2.
+    !!!!!!! integer, parameter :: n =3
+    !!!!!!! integer, parameter :: m =2  
+    ! x = [4.0,3.0,2.0]
+    ! !!!!!!! opt value
+    ! !!!!!!! x = [2.017526, 1.779968, 1.237558]
+    ! call func2 (n, m, x, f0val , df0dx, fval , dfdx)
+    ! print *, "f0val=", f0val, "fval=", fval
+    ! print '(A,I3,A,3F10.4,A,F10.7,A,F10.7,A,F10.7)', 'iter=', 0, ' , x= ', &
+    !    x, '-------,f0val= ', f0val, ',   fval1= ', fval(1), ',   fval2= ', &
+    !    fval(1)
```

Comment:
Please create two subroutines, `test_case_1` and `test_case_2` to at minimum
reduce the amount of commenting based switching to a single line:
```
call test_case_1
call test_case_2
```

But this should be converted to a pFUnit test
Please look into:
https://github.com/Goddard-Fortran-Ecosystem/pFUnit_demos